### PR TITLE
complete main feature

### PR DESCRIPTION
**Pull request by lexign**
1. 🔧 add casbinRule model ts file
2. ✨ add SequelizeAdapter class ts file
3. 🎨 declare global sequelize attributes
4. 📝 add index ts file for emport
5. ✏️ fix test typo
6. 📦 use yarn instead of npm

### CODE REVIEW

File: src/adapter.ts [hunk 1]
```diff
@@ -12,155 +12,187 @@
...
+      lineText += ', ' + line.v1;
     }
-
-    private close() {
-        this.sequelize.close();
+    if (line.v2) {
+      lineText += ', ' + line.v2;
     }
-
-    private createTable() {
```
Comment (hsluoyz) [inline src/adapter.ts:52]: Do not delete ``createTable()`` and ``dropTable()``

File: src/adapter.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018 The Casbin Authors. All Rights Reserved.
```
Comment (hsluoyz) [inline src/adapter.ts:1]: Do not modify here.

**Review by hsluoyz — COMMENTED** [review]
I have said about these in previous PR. It is still not fixed.

File: src/adapter.ts (hunk 1 above)
Comment (lexign) [reply src/adapter.ts:52]: There is a sync method === create table auto.

**Comment by hsluoyz** [comment]
Save this code for future use:

```typescript
// addPolicy adds a policy rule to the storage.
public async addPolicy(sec: string, ptype: string, rule: string[]): Promise<void> {
    const {casbinRule} = this;
    const line: any = this.savePolicyLine(ptype, rule);
    await casbinRule.create(line);
}

// removePolicy removes a policy rule from the storage.
public async removePolicy(sec: string, ptype: string, rule: string[]): Promise<void> {
    const {casbinRule} = this;
    const where: any = this.savePolicyLine(ptype, rule);
    await casbinRule.destroy({where});
}

// removeFilteredPolicy removes policy rules that match the filter from the storage.
public async removeFilteredPolicy(sec: string, ptype: string, fieldIndex: number, ...fieldValues: string[]): Promise<void> {
      const {casbinRule} = this;
      const where: any = {p_type: ptype};
      const sum: number = fieldIndex + fieldValues.length;
      let count: number = 0;
      if (fieldIndex <= count && sum > count) {
        where.v0 = fieldValues[count - fieldIndex];
      }
      count += 1;
      if (fieldIndex <= count && sum > count) {
        where.v1 = fieldValues[count - fieldIndex];
      }
      count += 1;
      if (fieldIndex <= count && sum > count) {
        where.v2 = fieldValues[count - fieldIndex];
      }
      count += 1;
      if (fieldIndex <= count && sum > count) {
        where.v3 = fieldValues[count - fieldIndex];
      }
      count += 1;
      if (fieldIndex <= count && sum > count) {
        where.v4 = fieldValues[count - fieldIndex];
      }
      count += 1;
      if (fieldIndex <= count && sum > count) {
        where.v5 = fieldValues[count - fieldIndex];
      }
      await casbinRule.destroy({where});
}
```

**Comment by hsluoyz** [comment]
The CI is ready now. New PRs can be tested automatically. You can make new PRs for further improvement.

**Pull request closed without merging**